### ISSUE TITLE
What's the best way to add vintage synthesizer waveforms?

### ISSUE THREAD

**Issue by octetta** [bwhitman/amy#38]
Would it be better to:

- create an additional `*_lutset.h` file for each?
- create PCM entries?

The first waveforms I'm thinking about are inspired from the Korg DW8000 and Ensoniq ESQ-1

(Korg's might be trickier, as there are different resolution waveforms per octave.)

**Comment by dpwe** [comment]
I'm not familiar with those synths, but I just read about the DWGS on the
DW8000's wikipedia page, and the lutset mechanism would be quite suitable,
if you have some source for the original cycle waveforms. (The point of
lutsets is to provide multiple lookup tables of a single waveform at
different sampling rates, to avoid aliasing at higher pitches, but you
could create these from a single, full-spectrum prototype by properly
anti-aliased downsampling).

There would need to be a new oscillator type that took the source lut
(index) as an argument.  It would look just like the triangle oscillator,
though.

How good/useful were these additional raw waveforms?  There's a lot more to
a piano than a single cycle of waveform!

  DAn.

**Comment by octetta** [comment]
I looked at the lutset code a little closer and understand your point about the DW8000/DWGS approach. ESQ-1 waves to my knowledge are single

As for the utility of said waveforms, I agree that they're very sonic-ly simple, compared to say the FM algorithm, or managed partials.

The value they might bring would be akin to what y'all did with DX-7 patches. I'd like to resurrect old DW8000 and ESQ-1 patches. I was in a synth band in the 80's and those two were my faves.

I've been gloming lots of my own ideas onto AMY, perhaps at an obtuse angle from what you and Brian have in mind.

I've pitched a lightning talk at a software conference in a few months to showcase me bolting AMY together with a MIDI controller and the Elixir/Erlang environment.

"Gosh, really got some nice toys here." - Roy Batty / Bladerunner
